test/Models/CsvCatalogItemSpec.js

```diff
@@ -437,6 +437,112 @@ describe('CsvCatalogItem with lat and lon', function() {
+                csvItem._minPix = Math.min.apply(null, pixelSizes);
+                csvItem._maxPix = Math.max.apply(null, pixelSizes);
+                // we don't want to be too prescriptive, but by default the largest object should be 150% normal, smallest is 50%, so 3x difference.
+                expect(csvItem._maxPix).toEqual(csvItem._minPix * 3);
+            }).then(function() {
+                var csvItem2 = new CsvCatalogItem(terria);
+                csvItem2._tableStyle = new TableStyle({scale: 10, scaleByValue: true });  // this time, apply it to all columns
+                csvItem2.url = 'test/csv/lat_lon_val.csv';
+                return csvItem2.load().yield(csvItem2);
+            }).then(function(csvItem2) {
+                var pixelSizes = csvItem2.dataSource.entities.values.map(function(e) { return e.point._pixelSize._value; });
+                var minPix = Math.min.apply(null, pixelSizes);
+                var maxPix = Math.max.apply(null, pixelSizes);
+                // again, we don't specify the base size, but x10 things should be twice as big as x5 things.
+                expect(maxPix).toEqual(csvItem._maxPix * 2);
+                expect(minPix).toEqual(csvItem._minPix * 2);
+            }).otherwise(fail).then(done);
+        });
+
+        it('uses correct defaults', function(done) {
+            // nullColor is passed through to the columns as well, if not overridden explicitly.
+            csvItem.url = 'test/csv/lat_lon_badvalue.csv';
+            csvItem._tableStyle = new TableStyle({
+                nullColor: '#A0B0C0',
+                columns: {
+                    value: {
+                        replaceWithNullValues: ['bad']
+                    }
+                }
+            });
+            var nullColor = new Color(160/255, 176/255, 192/255, 1);
+            csvItem.load().then(function() {
+                function cval(i) { return csvItem.dataSource.entities.values[i]._point._color._value; }
+                expect(cval(1)).toEqual(nullColor);
+            }).otherwise(fail).then(done);
+        });
+
+        it('supports name and nullColor with column ref by name', function(done) {
+            csvItem.url = 'test/csv/lat_lon_badvalue.csv';
+            csvItem._tableStyle = new TableStyle({
+                nullColor: '#123456',
+                columns: {
+                    value: {
+                        replaceWithNullValues: ['bad'],
+                        nullColor: '#A0B0C0',
+                        name: 'Temperature'
+                    }
+                }
+            });
+            var nullColor = new Color(160/255, 176/255, 192/255, 1);
+            csvItem.load().then(function() {
+                expect(csvItem.tableStructure.columns[2].name).toEqual('Temperature');
+                function cval(i) { return csvItem.dataSource.entities.values[i]._point._color._value; }
```

Comment:
Was this supposed to be separated out? Seems a bit weird to be defining this as a function (twice) when you could just go `csvItem.dataSource.entities.values[1]._point._color._value`